src/awkward/_v2/identifier.py

```diff
@@ -0,0 +1,113 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from __future__ import absolute_import
+
+import awkward as ak
+
+np = ak.nplike.NumpyMetadata.instance()
+
+
+class Identifier(object):
+    _numrefs = 0
+
+    @staticmethod
+    def newref():
+        out = Identifier._numrefs
+        Identifier._numrefs += 1
+        return out
+
+    def __init__(self, ref, fieldloc, data):
+        self._ref = ref
+        self._fieldloc = fieldloc
+        if not isinstance(fieldloc, dict) or not all(
+            isinstance(k, int) and isinstance(v, str) for k, v in fieldloc.items()
+        ):
+            raise TypeError("Identifier fieldloc must be a dict of int -> str")
+        self._nplike = ak.nplike.of(data)
+
+        self._data = self._nplike.asarray(data, order="C")
+        if len(self._data.shape) != 2:
+            raise TypeError("Identifer data must be 2-dimensional")
```

Comment:
```suggestion
            raise TypeError("Identifier data must be 2-dimensional")
```

Reply:
Sorry! We missed this while we were talking (and not looking at updates here). I got your issue and PR, and have set it to auto-merge. Thanks!